janus-idp/backstage-showcase
# fix(ui): RHIDP-2044 enable frontend plugin loading

## Description

This change removes the authentication requirement for the scalprum backend as it's used to serve static javascript content and related manifests.  This commit also adds identity support to the dynamic plugins info list frontend and migrates the backend to use the httpAuth service.

This change adds the new upstream guest authentication provider to the backend.  This provider is enabled by adding a "guest" entry to the `auth.providers` map.  This change also removes the dynamic-plugins-info-backend e2e test, as
it relies on an unauthenticated endpoint that isn't available anymore.  This functionaly is already covered by the UI e2e test and the component's unit test.

## Which issue(s) does this PR fix

- Fixes [RHIDP-2044](https://issues.redhat.com/browse/RHIDP-2044)
- Fixes [RHIDP-2043](https://issues.redhat.com/browse/RHIDP-2043)

## PR acceptance criteria

Please make sure that the following steps are complete:


## How to test changes / Special notes to the reviewer
This PR also adds the new upstream guest authentication provider.  You should be able to log into the app using this provider (or github/keycloak etc), click on the Administration tab and see the list of available plugins as a regular user.  There should be no HTTP 401 errors logged to the browser console when accessing the scalprum backend.  To use the guest authentication provider it now must be explicitly enabled in the auth provider map like so:

```yaml
# When running locally with "yarn start" only this is needed
auth:
  providers:
    guest: {}

# When running against a built image:
auth:
  providers:
    guest:
      dangerouslyAllowOutsideDevelopment: true
```
